### DISCUSSION TITLE
For which architectures is Wolfi available?

### DISCUSSION THREAD

**Discussion by kastl-ars** [wolfi-dev/os#21241] [Q&A] [answered]
I might just be missing the right search terms or be overlooking this, but for which architectures is WolfiOS or its packages available? I know amd64 and arm64, but what about others? Is there a list somewhere?

Thanks in advance!

Johannes

**Comment by xnox** [comment, marked as answer]
WolfiOS is currently available for x86-64 (amd64) and aarch64 (arm64). Current target for WolfiOS is to run as containers in the public cloud and these are the two main architectures offered by most public clouds.

Which architectures are you interested in?

**Reply by kastl-ars** [reply]
I agree those are the main ones. armv7l (think Raspberry PI 2) came to my mind, running a small container on one of those would be nice (even though I do not know how many people actually would use that.

I would propose to add a sentence regarding the architectures to the Wolfi README or FAQ, for the next one looking for this information...

**Reply by xnox** [reply]
@kastl-ars actual armv7l Raspberry Pi's do not exist for a few years anymore, they have replaced most SKUs with arm64 processors whilst still selling/marketing them as armv7.  As in even an 8 year old Raspberry Pi can successfully run Chainguard Images & Wolfi Images (as well as Apple silicon Macbooks, and Windows on Arm laptops). The 10 years old Raspberry Pis do not have enough ram to start host OS + docker/k8s to run an image, so one gets OOM as soon as one executes more than a few processes thus not useful at all.

Note security support for armv7 is severely limited across the ecosystem, whilst Ubuntu & Debian still provide armhf builds, majority of upstream software projects have dropped support for it or do not support it. It is also the case for compilers / kernels / toolchains especially w.r.t. hardening and security features.

If at all possible please use arm64 capable Raspberry Pis.
